### DISCUSSION TITLE
Sentence splitting seems almost random on Spacy 3.0 & en_core_web_lg

### DISCUSSION THREAD

**Discussion by adam-ra** [explosion/spaCy#7035] [Help: Coding & Implementations] [answered]
## How to reproduce the behaviour
```
In [1]: import spacy; nlp = spacy.load('en_core_web_lg')                        

In [2]: list(nlp("tired all of the time").sents)                                
Out[2]: [tired, all of the time]

In [3]: list(nlp("Severe pain, after trauma").sents)                            
Out[3]: [Severe pain, ,, after trauma]

In [4]: list(nlp("headache when bending head").sents)                           
Out[4]: [headache, when bending head]

In [5]: list(nlp("jaundice after stress").sents)                                
Out[5]: [jaundice after stress]

In [6]: list(nlp("jaundice after increased stress").sents)                      
Out[6]: [jaundice, after increased stress]

```

## Your Environment
Python 3.8.5, Ubuntu 20.04, 5.4.0-65-generic
spacy==3.0.1
en-core-web-lg @ https://github.com/explosion/spacy-models/releases/download/en_core_web_lg-3.0.0/en_core_web_lg-3.0.0-py3-none-any.whl

**Comment by svlandeg** [comment]
Hi! I'm going to convert this issue to the discussions as I think it's more suited for the forum than for the issue tracker. You may get a notification that this thread was closed - but really it was just moved and is still open on the forum :-)

**Comment by svlandeg** [comment]
What I assume is the problem here, is the fact that you're applying the model on texts that are quite unlike the type of texts it has seen before. The model was trained on generic texts from OntoNotes 5.0. Most of these articles will contain full sentence with enough syntactic clues to deduce the sentences. In contrast, the texts you're applying this model on seem to consist more of phrases without much grammar.

Consider for instance:
```
    import spacy
    nlp = spacy.load('en_core_web_lg')

    sents = nlp("He is tired all of the time. Has severe pain, after trauma, and headache when bending head. Jaundice occurs after increased stress").sents
    for sent in sents:
        print(sent)
```

Which gives you 

> He is tired all of the time.
> Has severe pain, after trauma, and headache when bending head.
> Jaundice occurs after increased stress

So it's definitely not random - it's just not well suited to your challenge, unfortunately.

One thing to consider is to [(re)train](https://spacy.io/usage/training) a model on your specific dataset & domain, to ensure it'll work more like you'd expect on your specific use-case.

**Reply by adam-ra** [reply]
Well, obviously the material used to train will influence the output and best performance would be observed on full sentences.
But still, splitting valid English noun phrases on prepositions seems pretty odd.

But please consider tese two points:

1. Here the sentence splitter actually splits noun phrases inside. This is quite nasty to any application. For instance it sees two sentences in “tired all of the time”, similarly in  “jaundice after stress”. We have dozens of similar examples more. 
2. In Spacy 2.3.2 and model with the same name (the large one) none of these splittings inside phrases appear (none of those dozens). I really suspect there's some bug here

**Reply by honnibal** [reply]
I do think this could point to a bug, the problems here are quite surprising. I changed the way the `Break` transition works for 3.0, so I'll look into what's going on here.

**Reply by adam-ra** [reply]
Thanks! I'd really appreciate a look into that :)

**Comment by honnibal** [comment, marked as answer]
Confirming this is a bug. Here's the transition history for `Severe pain , after trauma`:

```
Transitions: ['S', 'L-amod', 'S', 'S', 'S', 'R-pobj', 'D', 'D', 'D', 'D']
Sentences:
    Severe pain
    ,
    after trauma
```

We shouldn't be getting a sentence-break there, given that the Break isn't predicted.

* Write-up: #7056
* Fix: #7057

**Reply by adam-ra** [reply]
Thanks a lot for a quick investigation and bug fix! :bow:
